Service/FilterService.php

```diff
@@ -0,0 +1,164 @@
+        LoggerInterface $logger = null
+    ) {
+        $this->dataManager = $dataManager;
+        $this->filterManager = $filterManager;
+        $this->cacheManager = $cacheManager;
+        $this->logger = $logger ? $logger : new NullLogger();
+    }
+
+    /**
+     * @param string $path
+     * @param string $filter
+     * @param string $resolver
+     * @throws \Liip\ImagineBundle\Exception\Binary\Loader\NotLoadableException
+     * @throws \Liip\ImagineBundle\Exception\Imagine\Filter\NonExistingFilterException
+     */
+    public function createFilteredImage($path, $filter, $resolver = null)
+    {
+        if ($this->cacheManager->isStored($path, $filter, $resolver)) {
+            return;
+        }
+
+        $filteredBinary = $this->createFilteredBinary(
+            $path,
+            $filter
+        );
+
+        $this->cacheManager->store(
+            $filteredBinary,
+            $path,
+            $filter,
+            $resolver
+        );
+    }
+
+    /**
+     * @param string $path
+     * @param string $filter
+     * @param string $resolver
+     * @param array $runtimeFilters
+     * @throws \Liip\ImagineBundle\Exception\Binary\Loader\NotLoadableException
+     * @throws \Liip\ImagineBundle\Exception\Imagine\Filter\NonExistingFilterException
+     */
+    public function createFilteredImageWithRuntimeFilters($path, $filter, array $runtimeFilters = array(), $resolver = null)
+    {
+        $runtimePath = $this->cacheManager->getRuntimePath($path, $runtimeFilters);
+        if ($this->cacheManager->isStored($runtimePath, $filter, $resolver)) {
+            return;
+        }
+
+        $filteredBinary = $this->createFilteredBinary(
+            $path,
+            $filter,
+            $runtimeFilters
+        );
+
+        $this->cacheManager->store(
+            $filteredBinary,
+            $runtimePath,
+            $filter,
+            $resolver
+        );
+    }
+
+    /**
+     * @param string $path
+     * @param string $filter
+     * @param string $resolver
+     * @return string
+     */
+    public function getUrlOfFilteredImage($path, $filter, $resolver = null)
+    {
+        return $this->cacheManager->resolve($path, $filter, $resolver);
+    }
+
+    /**
+     * @param string $path
+     * @param string $filter
+     * @param string $resolver
+     * @param array $runtimeFilters
+     * @return string
+     */
+    public function getUrlOfFilteredImageWithRuntimeFilters($path, $filter, array $runtimeFilters = array(), $resolver = null)
+    {
+        $runtimePath = $this->cacheManager->getRuntimePath($path, $runtimeFilters);
+        return $this->cacheManager->resolve($runtimePath, $filter, $resolver);
+    }
+
+    /**
+     * @param string $path
+     * @param string $filter
+     * @param array|null $runtimeFilters
+     * @return \Liip\ImagineBundle\Binary\BinaryInterface
+     * @throws \Liip\ImagineBundle\Exception\Imagine\Filter\NonExistingFilterException
+     */
```

Comment:
All doc blocks need their CS fixes such that they are ordered `param`, `throws`, `return`, each type is separated by a two newlines, and have their `param` variables aligned. We also use short class names, even if they aren't used in the current file a and require an additional `use` statement.
```php
/**
 * @param string     $path
 * @param string     $filter
 * @param array|null $runtimeFilters
 *
 * @throws \Liip\ImagineBundle\Exception\Imagine\Filter\NonExistingFilterException
 *
 * @return \Liip\ImagineBundle\Binary\BinaryInterface
 */
```